libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -340,7 +341,13 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
       quietMode: isQuietMode,
       debugMode: this.parser.isDebug,
       parallelism,
-      changedProjectsOnly
+      changedProjectsOnly,
+      beforeExecuteOperations: async (records: Map<Operation, OperationExecutionRecord>) => {
+        await this.hooks.beforeExecuteOperations.promise(records);
+      },
+      onOperationStatusChanged: async (record: OperationExecutionRecord) => {
```

Comment:
Since this particular hook is purely informational and we aren't waiting for this callback to complete, we should be clear about that in the hook type and make it a `SyncHook`, and use `(record: OperationExecutionRecord) => this.hooks.onOperationStatusChanged.call(record)`

Reply:
yeah, I should think about which one needs to wait for completion before implementing these two hooks